client/extensions/woocommerce/state/sites/settings/email/reducer.js

```diff
@@ -22,39 +22,55 @@ import {
 	WOOCOMMERCE_EMAIL_SETTINGS_SUBMIT_FAILURE,
 	WOOCOMMERCE_EMAIL_SETTINGS_INVALID_VALUE,
 } from 'woocommerce/state/action-types';
+
+const process_data = data => {
+	const options = {};
+	const fromAddress = filter( data, {
+		group_id: 'email',
+		id: 'woocommerce_email_from_address',
+	} );
+	const defaultEmail = isEmpty( fromAddress ) ? '' : fromAddress[ 0 ].default;
+	data.forEach( function( option ) {
+		setWith(
+			options,
+			[ option.group_id, option.id ],
+			{
+				value: option.value,
+				default: option.default,
+			},
+			Object
+		);
+	} );
+
+	forEach( [ 'email_new_order', 'email_cancelled_order', 'email_failed_order' ], key => {
+		const option = get( options, [ key ], {} );
+		const def = get( option, [ 'recipient', 'default' ], false ) || defaultEmail;
+		const value = get( option, [ 'recipient', 'value' ], '' );
+		const noValue = value === '';
+		const updatedValue = noValue ? def : value;
```

Comment:
This section seems like it could be reduced a bit or made a bit clearer, and this seems like a good place to have some tests as well. A few of these variables are also only used once.

i.e. `updatedValue` is only ever used if `noValue` is true, so a few lines down 

```
if ( enabled && noValue ) {
    options[ key ].recipient.value = updatedValue;
}
```

could just be 


```
if ( enabled && noValue ) {
    options[ key ].recipient.value = def; // default might be a better name
}
```

Reply:
I have made it even simpler.